src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNIndexCache.java

```diff
@@ -147,6 +162,35 @@ public CacheStats getStats() {
         return cache.stats();
     }
 
+    /**
+     * Get the Elasticsearch indices currently loaded into the Cache
+     *
+     * @return Set containing all of the Elasticsearch indices in the cache
+     */
+    public Map<String, Map<String, Object>> getIndicesCacheStats() {
+        Map<String, Map<String, Object>> statValues = new HashMap<>();
+        String indexName;
+        for (Map.Entry<String, KNNIndexCacheEntry> index : cache.asMap().entrySet()) {
+            indexName = index.getValue().getEsIndexName();
+            statValues.putIfAbsent(indexName, new HashMap<>());
+
+            statValues.get(indexName).putIfAbsent(GRAPH_COUNT, 0);
+            statValues.get(indexName).put(GRAPH_COUNT, ((Integer) statValues.get(indexName).get(GRAPH_COUNT)) + 1);
```

Comment:
statValues.get(indexName).put(GRAPH_COUNT, ((Integer) statValues.get(indexName).getorDefault(GRAPH_COUNT, 0)) + 1)

Reply:
> Thanks for enriching the stats. Few other stats which might be helpful as well
> 
>     * graph_memory_usage_percentage in terms of total cache capacity at node level
> 
>     * index_graph_memory_usage_percentage at index level for that node in terms of total cache capacity
> 
>     * index_graph_loads at index level which counts the number of graph loads for an index in that node. This helps in identifying the hot indices in the cache. We could add a counter variable in the KNNIndexCacheEntry and increment it when ever the entry is accessed.

I think the first two make sense. I am not sure on the last one. Isn't the number of graphs per index somewhat analogous to index_graph_loads?

Reply:
Ah right. Third one seem to be already covered. You could ignore the third one.